Update cost scaling heuristic to vary speed linearly with distance (#2164)
ros-planning/navigation2: nav2_regulated_pure_pursuit_controller/README.md

| `max_allowed_time_to_collision` | The time to project a velocity command to check for collisions | 
| `use_regulated_linear_velocity_scaling` | Whether to use the regulated features for curvature | 
| `use_cost_regulated_linear_velocity_scaling` | Whether to use the regulated features for proximity to obstacles | 

> | `cost_scaling_dist` | The minimum distance from an obstacle to trigger the scaling of linear velocity, if `use_cost_regulated_linear_velocity_scaling` is enabled. The value set should be smaller than the `inflation_radius` set in the inflation layer of costmap, since inflation is used to compute the distance from obstacles | 

Reviewer: ```suggestion
| `cost_scaling_dist` | The distance from an obstacle to trigger the scaling of linear velocity, if `use_cost_regulated_linear_velocity_scaling` is enabled. The value set should be smaller or equal to the `inflation_radius` set in the inflation layer of costmap, since inflation is used to compute the distance from obstacles | 
```

> | `cost_scaling_gain` | A multiplier used to further scale the speed when within the specified `cost_scaling_dist` | 

Reviewer: ```suggestion
| `cost_scaling_gain` | A multiplier gain, which should be > 1.0, used to further scale the speed when an obstacle is within `cost_scaling_dist`. Higher value reduces speed more quickly. | 
```

Author: Currently gain is being multiplied with the velocity so it should be <1.
I realise gain means amplification so should I just multiply the velocity with the inverse of gain or we could use some other term in place of gainz

Reviewer: Oh, sorry - ok, update to say that then :-)

Reviewer: dont worry about changing the formulation, either way you need to tell people > 1 or < 1, so lets just go with what we got
